platform/core/src/capabilities/MutationCapability.js

```diff
@@ -141,11 +141,17 @@ define(
                 if (mutationResult !== false) {
                     // Copy values if result was a different object
                     // (either our clone or some other new thing)
-                    if (model !== result) {
+                    let modelHasChanged = JSON.stringify(model) !== JSON.stringify(result);
```

Comment:
Comparing stringified objects will probably work just fine in this case, but there are edge cases where it will fail as a general method for comparing objects. If the order of properties changes, but the values of the properties remain the same then a string comparison will detect a change, even though the objects are functionally equivalent. This might happen if an attribute is deleted, and then assigned the same value again.

Again, for this purpose it probably doesn't matter, however we might consider using `_.equals` as we have in other parts of the code, which will handle this edge case.

Reply:
Done.